playgame.py

```diff
@@ -116,35 +116,51 @@ def prep_logger(log_dir, verbose, log_stderr, count):
         logger.addHandler(efh)
         efh.setFormatter(formatter)
 
-def parse_args():
+def parse_args(args):
     usage = 'plays games of Hanabi using the listed players'
 
     parser = argparse.ArgumentParser(description = usage)
 
     #Positional arguments
-    parser.add_argument('players', nargs = '+',
+    parser.add_argument('players',
+                        nargs = '+',
                         help = 'the players that will play Hanabi. First 5 will play unless in tournament mode')
 
     #Optional arguments
     parser.add_argument('-s', '--seed',
-                        default = int(round(time.time()*1000)), type = int,
+                        default = int(round(time.time()*1000)),
+                        type = int,
                         help = 'a specific seed for shuffling the deck')
-    parser.add_argument('-r', '--variant', type = int, choices = [1, 2, 3],
+    parser.add_argument('-r', '--variant',
+                        type = int,
+                        choices = [1, 2, 3],
                         default = 0,
                         dest = 'variant',
                         help = 'play the selected variant')
-    parser.add_argument('-t', '--tournament', dest = 'is_tournament',
-                        action = 'store_true',
-                        help = 'run 2 player games for all combinations of players (no repeats)')
-    parser.add_argument('-v', '--verbose', dest = 'verbose',
+    parser.add_argument('-v', '--verbose',
+                        dest = 'verbose',
                         action = 'store_true',
                         help = 'log moves and game state as game is played')
-    parser.add_argument('-l', '--log_dir', dest = 'log_dir', default = None,
+    parser.add_argument('-l', '--log_dir',
+                        dest = 'log_dir',
+                        default = None,
                         help = 'save logs to file')
-    parser.add_argument('-e', '--log_stderr', dest = 'log_stderr',
+    parser.add_argument('-e', '--log_stderr',
+                        dest = 'log_stderr',
                         help = 'log errors to file')
 
-    return parser.parse_args()
+    tournament_group = parser.add_argument_group('tournament arguments')
```

Comment:
so groups...
I want to not include tournament things in the single-game namespace, so I was playing around with that. Tried:
1. Using subparsers. I had to duplicate the shared params and that looked terrible
2. Groups. Now these params are logically separated, but don't have the hierarchical checks I'm hoping for. The idea is if per_round is specified, you should be in tournament mode

Reply:
I think subparsers could work if the shared params were defined on a common parent parser. [Here](https://stackoverflow.com/questions/7498595/python-argparse-add-argument-to-multiple-subparsers) and [here](https://stackoverflow.com/questions/33645859/how-to-add-common-arguments-to-argparse-subcommands/33646419) are some examples.

I think it might change the interface quite a bit. From what I can see, you'd need to do something like
`python playgame.py single [options]`
or
`python playgame.py tournament [options]`

If you don't like that route, another option could be to add a check for the condition you want explicitly, like [this example](https://stackoverflow.com/questions/19414060/argparse-required-argument-y-if-x-is-present).

I think it'd end up something like this:
```python
if args.per_round and args.is_tournament is None:
    parser.error("players per game only valid in tournament mode")
```

Reply:
Thanks for the insight. Huge fan of the parent parser option. Stops playgame from also doing parser logic